spec/grover/processor_spec.rb

```diff
@@ -579,6 +603,46 @@
         it { expect(pdf_text_content).to eq "#{date} http://www.example.net/foo/bar 1/1" }
       end
 
+      context 'when wait for function option is specified with options' do
+        let(:url_or_html) do
+          <<-HTML
+            <html>
+              <body></body>
+
+              <script>
+                setTimeout(function() {
+                  document.body.innerHTML = '<h1 id="test">Hey there</h1>';
+                }, 500);
+              </script>
+            </html>
+          HTML
+        end
+        let(:options) do
+          basic_header_footer_options.merge(
+            'waitForFunction' => 'document.getElementById("test") !== null',
+            'waitForFunctionOptions' => { "polling": 50, "timeout": 750 }
+          )
+        end
+        let(:date) { Date.today.strftime '%-m/%-d/%Y' }
+
+        it { expect(pdf_text_content).to eq "#{date} Hey there http://www.example.net/foo/bar 1/1" }
+
+        context 'when waiting for function takes too long' do
+          let(:options) do
+            basic_header_footer_options.merge(
+              'waitForFunction' => 'document.getElementById("test") !== null',
+              'waitForFunctionOptions' => { "polling": 50, "timeout": 250 }
```

Comment:
btw I would suggest as a means to avoid duplicating the setup, and to making it clearer what exactly is different about this content.. instead put the timeout value into a let variable up above, then this would just become: 
```
let(:wait_function_timeout) { 250 }
```

(and the bit up above being)
```
let(:options) do
  basic_header_footer_options.merge(
    'waitForFunction' => 'document.getElementById("test") !== null',
    'waitForFunctionOptions' => { "polling": 50, "timeout": wait_function_timeout }
  )
end
let(:wait_function_timeout) { 750 }
``